### PR TITLE
[HEVCD] Refactoring of errors handling

### PR DESCRIPTION
**Pull request by IlyaSergeyev**
To make debugging easier

### CODE REVIEW

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp [hunk 1]
```diff
@@ -373,7 +352,7 @@ mfxStatus VideoDECODEH265::Init(mfxVideoParam *par)
     if (m_platform != m_core->GetPlatformType())
```
Comment (mgonchar) [inline _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp]: we have ```MFX_CHECK_WITH_ASSERT``` you can replace this whole block with this macro

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp [hunk 2]
```diff
@@ -445,7 +414,7 @@ mfxStatus VideoDECODEH265::Reset(mfxVideoParam *par)
...
     {
         VM_ASSERT(m_platform == MFX_PLATFORM_SOFTWARE);
-            return MFX_ERR_UNSUPPORTED;
+        MFX_RETURN(MFX_ERR_UNSUPPORTED);
```
Comment (mgonchar) [inline _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp]: same

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp [hunk 3]
```diff
@@ -462,8 +431,7 @@ mfxStatus VideoDECODEH265::Close(void)
     MFX_AUTO_LTRACE(MFX_TRACE_LEVEL_API, "VideoDECODEH265::Close");
     UMC::AutomaticUMCMutex guard(m_mGuard);
 
-    if (!m_isInit || !m_pH265VideoDecoder.get())
-        return MFX_ERR_NOT_INITIALIZED;
+    MFX_CHECK(m_isInit && m_pH265VideoDecoder.get(), MFX_ERR_NOT_INITIALIZED);
```
Comment (mgonchar) [inline _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp]: and drop ```.get()``` :)

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp [hunk 4]
```diff
@@ -719,7 +683,7 @@ mfxStatus VideoDECODEH265::QueryIOSurf(VideoCORE *core, mfxVideoParam *par, mfxF
     if (platform != core->GetPlatformType())
     {
         VM_ASSERT(platform == MFX_PLATFORM_SOFTWARE);
-            return MFX_ERR_UNSUPPORTED;
+        MFX_RETURN(MFX_ERR_UNSUPPORTED);
```
Comment (mgonchar) [inline _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp]: same

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp [hunk 5]
```diff
@@ -1306,7 +1258,7 @@ mfxStatus VideoDECODEH265::DecodeFrame(mfxFrameSurface1 *surface_out, H265Decode
         if (!pFrame)
         {
             VM_ASSERT(false);
-            return MFX_ERR_NOT_FOUND;
+            MFX_RETURN(MFX_ERR_NOT_FOUND);
```
Comment (mgonchar) [inline _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp]: same

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp (hunk 1 above)
Comment (IlyaSergeyev) [reply _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp]: MFX_CHECK_WITH_ASSERT has one condition for assert and check, but they are different here

Comment (mgonchar) [reply _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp]: oh, I see

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp [hunk 6]
```diff
@@ -218,28 +214,22 @@ mfxStatus VideoDECODEH265::Init(mfxVideoParam *par)
     m_vPar.mfx.NumThread = (mfxU16)CalculateNumThread(par, m_platform);
 
     if (MFX_PLATFORM_SOFTWARE == m_platform)
```
Comment (mgonchar) [inline _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp]: MFX_CHECK

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp [hunk 7]
```diff
@@ -616,14 +581,12 @@ mfxStatus VideoDECODEH265::DecodeHeader(VideoCORE *core, mfxBitstream *bs, mfxVi
 
     if (umcRes == UMC::UMC_ERR_NOT_ENOUGH_DATA)
```
Comment (mgonchar) [inline _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp]: MFX_CHECK

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp (hunk 6 above)
Comment (IlyaSergeyev) [reply _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp]: @dmitry-gurulev comment, please

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp (hunk 7 above)
Comment (IlyaSergeyev) [reply _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp]: I don't want to see it in trace, it's not an error actualy

Comment (mgonchar) [reply _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp]: agree

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp (hunk 6 above)
Comment (mgonchar) [reply _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp]: Checked CS code base. Would be hard to align if we will use macro here

**Pull request merged**